hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java

```diff
@@ -60,4 +89,62 @@ public TransactionMetadata preHandle(
     public void handle(@NonNull final TransactionMetadata metadata) {
         throw new UnsupportedOperationException("Not implemented");
     }
+
+    /* --------------- Helper methods --------------- */
+
+    /**
+     * Validates the collector key from the custom fees.
+     *
+     * @param meta given transaction metadata
+     * @param customFeesList list with the custom fees
+     */
+    private void addCustomFeeCollectorKeys(
+            SigTransactionMetadataBuilder meta, final List<CustomFee> customFeesList) {
+
+        for (final var customFee : customFeesList) {
+            final var collector = customFee.getFeeCollectorAccountId();
+
+            /* A fractional fee collector and a collector for a fixed fee denominated
+            in the units of the newly created token both must always sign a TokenCreate,
+            since these are automatically associated to the newly created token. */
+            if (customFee.hasFixedFee()) {
+                final var fixedFee = customFee.getFixedFee();
+                final var alwaysAdd =
+                        fixedFee.hasDenominatingTokenId()
+                                && fixedFee.getDenominatingTokenId().getTokenNum() == 0L;
+                addAccount(meta, collector, alwaysAdd);
+            } else if (customFee.hasFractionalFee()) {
+                meta.addNonPayerKey(collector, INVALID_CUSTOM_FEE_COLLECTOR);
+            } else {
+                final var royaltyFee = customFee.getRoyaltyFee();
+                var alwaysAdd = false;
+                if (royaltyFee.hasFallbackFee()) {
+                    final var fFee = royaltyFee.getFallbackFee();
+                    alwaysAdd =
+                            fFee.hasDenominatingTokenId()
+                                    && fFee.getDenominatingTokenId().getTokenNum() == 0;
+                }
+                addAccount(meta, collector, alwaysAdd);
+            }
+        }
+    }
+
+    /**
+     * Signs the metadata or adds failure status.
+     *
+     * @param meta given transaction metadata
+     * @param collector the ID of the collector
+     * @param alwaysAdd if true, will always add the key
+     */
+    private void addAccount(
+            final SigTransactionMetadataBuilder meta,
+            final AccountID collector,
+            final boolean alwaysAdd) {
+
+        if (alwaysAdd) {
+            meta.addNonPayerKey(collector);
```

Comment:
I think you should also pass in `INVALID_CUSTOM_FEE_COLLECTOR` here right ?

Reply:
Oh, yes, great catch --- fixed this and added a new scenario to test it